Automattic/wp-calypso
# Design Picker: Upgrade Modal to support Marketplace Themes

Fixes https://github.com/Automattic/dotcom-forge/issues/2956

## Proposed Changes

Update the Upgrade Modal on Design Picker to support Marketplace Themes. Styles on: RcxiuWYjZGkFRdlWL7L02m-fi-754:221984#540834845

| Desktop  | Mobile |
| ------------- | ------------- |
|<img width="1238" alt="CleanShot 2023-08-31 at 12 51 41@2x" src="https://github.com/Automattic/wp-calypso/assets/5039531/b154c7f8-5d03-42df-8b90-86bbed69050c">|<img width="354" alt="CleanShot 2023-08-31 at 12 52 06@2x" src="https://github.com/Automattic/wp-calypso/assets/5039531/9796afd7-3bbf-4608-946f-5bd91ece97bb">|


## Testing Instructions

* Apply the diff D118900-code to your sandbox and direct requests to the sandbox
* Go to the Design Picker page with a Marketplace theme selected `/setup/update-design/designSetup?siteSlug=:site&theme=:theme_slug`. Theme slugs such as `makoney` and `olsen-fse` can be used.
* Click on `Unlock Theme`
* You should see the modal with the updated layout.
* Test this flow with a site without plans and a site with a business plan
* On the Business plan site, you shouldn't see the business plan described on the modal


## Pre-merge Checklist
